packages/ckb-indexer/README.md

```diff
@@ -0,0 +1,456 @@
+CKB indexer used in lumos framework. Might be possible for independent usage as well. It is based on  [ckb-indexer](https://github.com/nervosnetwork/ckb-indexer) and export more interface.
+
+The indexer is designed to consume from the following sources:
+
+- Run on the web side.
+- Queries of CKB's RPC.
+
+## **Usage**
+
+### **Indexer**
+
+```jsx
+const { Indexer } = require("@ckb-lumos/ckb-indexer");
+const nodeUri = "https://testnet.ckb.dev/rpc";
+const indexUri = "https://testnet.ckb.dev/indexer";
+const indexer = new Indexer(indexUri, nodeUri);
+```
+
+### **CellCollector**
+
+To query existing cells, you can create a CellCollector:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x0000000000000000000000000000000000000000000000000000000000000000",
+    hash_type: "data",
+    args: "0x62e907b15cbf27d5425399ebf6f0fb50ebb88f18",
+  },
+});
+
+for await (const cell of cellCollector.collect()) {
+  console.log(cell);
+}
+```
+
+You can also specify both `lock` and `type` script:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    args: "0x92aad3bbab20f225cff28ec1d856c6ab63284c7a",
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+  },
+  type: {
+    args: "0x",
+    code_hash:
+      "0x82d76d1b75fe2fd9a27dfbaa65a039221a380d76c926f378d3f81cf3e7e13f2e",
+    hash_type: "type",
+  },
+});
+```
+
+Range query for cells between given block_numbers is supported:
+
+```jsx
+cellCollector = new CellCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7df323",
+  },
+  fromBlock: "0x225510", // "0x" + 2250000n.toString(16)
+  toBlock: "0x225ce0", // "0x" + 2252000n.toString(16)
+});
+
+for await (const cell of cellCollector.collect()) {
+  console.log(cell);
+}
+```
+
+It will fetch cells between `[fromBlock, toBlock]`, which means both `fromBlock` and `toBlock` are included in query range.
+
+Page jump when query cells is supported:
```

Comment:
```suggestion
Skip a certain number of query results, e.g. the below code snippet means it would skip the first 100 cells and return from the 101st one
```